third_party/winuwp_h264/H264Encoder/H264Encoder.cc

```diff
@@ -329,14 +334,32 @@ int WinUWPH264EncoderImpl::Encode(
   const VideoFrame& frame,
   const CodecSpecificInfo* codec_specific_info,
   const std::vector<FrameType>* frame_types) {
+  if (!inited_) {
+    return WEBRTC_VIDEO_CODEC_UNINITIALIZED;
+  }
+
   {
-      rtc::CritScope lock(&crit_);
-      if (!inited_) {
-      return -1;
+    auto frame_buffer = frame.video_frame_buffer();
+    auto cur_width = frame_buffer->width();
+    auto cur_height = frame_buffer->height();
+    auto now = rtc::TimeMillis();
+
+    rtc::CritScope lock(&crit_);
+    // Reset the encoder configuration if necessary.
+    bool res_changed = cur_width != (int)width_ || cur_height != (int)height_;
+    bool time_to_change_rate =
```

Comment:
This is not a time; can we rename to something a bit more explicit like `need_rate_change`?

Reply:
Yeah this is a leftover from a previous version.